torchvision/models/detection/retinanet.py

```diff
@@ -100,7 +100,8 @@ def compute_loss(self, targets, head_outputs, matched_idxs):
             foreground_idxs_per_image = matched_idxs_per_image >= 0
             num_foreground = foreground_idxs_per_image.sum()
             # no matched_idxs means there were no annotations in this image
-            if False:#matched_idxs_per_image.numel() == 0:
+            # TODO: enable support for images without annotations that works on distributed
```

Comment:
What is the problem with images without annotations on distributed?

Reply:
There might be cases where in one GPU no images had annotations, while in the other there were. In this case, there will be part of the computation graph which would not be executed in one GPU, leading to synchronization issues (and even deadlocks).

So for now I'm disabling support for this to move forward, and we will enable it in a later PR (after the release)

Reply:
Note that we are not passing `find_unused_params` to DDP, which you were probably using (because I had to change something else in the `resnet_fpn_backbone` for it to work). With `find_unused_params=True`, this might not be an issue, but I prefer to be on the safer side and let all computation graphs to be the same on every GPU